ptypes/regen.sh

```diff
@@ -22,6 +24,26 @@ function die() {
   exit 1
 }
 
+function which_sed() {
+  local uname_s="$(uname -s)"
+  case "${uname_s}" in
+    Darwin)
+      if ! which gsed > /dev/null; then
+        die "must install gsed on a mac, try brew install gnu-sed"
```

Comment:
ewww.

What GNU-isms are we actually using below? Can we fix those instead, and leave it as invoking plain "sed"?